arctic/serialization/numpy_arrays.py

```diff
@@ -120,28 +77,64 @@ def docify(self, df):
         df:  DataFrame
             The Pandas DataFrame to encode
         """
-        doc = SON({DATA: {}}, c=[str(c) for c in df.columns])
+        doc = SON({DATA: {}, METADATA: {}})
+
+        dtypes = {}
+        masks = {}
+        lengths = {}
+        columns = []
+        data = Binary(b'')
+        start = 0
+
         for c in df:
-            meta = {NAME: str(c)}
             try:
-                doc[DATA][str(c)] = self.converter.docify(df[c].values, meta)
+                columns.append(str(c))
+                arr, mask = self._convert_types(df[c].values)
+                dtypes[str(c)] = arr.dtype.str
+                if mask is not None:
+                    masks[str(c)] = Binary(compress(mask.tostring()))
```

Comment:
did you mean i should compress mask and arr in paralle? or did you mean i should save off all the arrs and compress all of them in parallel?